src/Tally.jl

```diff
@@ -262,42 +260,70 @@ end
 #
 ################################################################################
 
-@inline Base.iterate(T::TallyT) = iterate(T.data)
+Base.keys(T::TallyT) = T.keys
 
-@inline function Base.iterate(T::TallyT, st)
-  x = Base.iterate(T.data, st)
-  return (x === nothing ? nothing : x)
-end
+Base.values(T::TallyT) = T.values
 
 Base.eltype(::Type{TallyT{T}}) where {T} = Pair{T, Int}
 
-Base.length(T::TallyT) = length(T.data)
+Base.length(T::TallyT) = length(T.keys)
+
+function Base.iterate(T::TallyT, st = 1)
+  return (st > length(T) ? nothing : (T.keys[st] => T.values[st], st + 1))
+end
 
 function _push!(T::TallyT, x)
-  D = T.data
-  fl, j = _has_key(x, D, T.by, T.equivalence)
+  K = T.keys
+  V = T.values
+  fl, j = _has_key(x, K, T.by, T.equivalence)
   if fl
-    D[j] = (D[j][1] => D[j][2] + 1)
+    V[j] = V[j] + 1
   else
-    push!(D, (x => 1))
+    push!(K, x)
+    push!(V, 1)
   end
   return T
 end
 
+function Base.sort!(T::TallyT; rev = true, kwargs...)
+  p = sortperm(T.values; rev = rev, kwargs...)
+  permute!(T.keys, p)
+  permute!(T.values, p)
+  return T
+end
+
 function Base.push!(T::TallyT, x)
   _push!(T, x)
-  sort!(T.data, by = x -> x[2], rev = true)
+  sort!(T)
   return T
 end
 
 function Base.append!(T::TallyT, it)
   for x in it
     _push!(T, x)
   end
-  sort!(T.data, by = x -> x[2], rev = true)
+  sort!(T)
   return T
 end
 
+function Base.get(T::TallyT, key, default)
```

Comment:
Is this required for the dictionary interface? I would have thought that `getindex` is enough? Or do we get more methods for free when implementing `get`?

Reply:
for AbstractDictionaries, `getindex` calls `get`, so we only need that